lib/smart_city/dataset.ex

```diff
@@ -103,11 +103,27 @@ defmodule SmartCity.Dataset do
         technical: Technical.new(tech)
       })
 
-    {:ok, struct}
+    errors = return_errors(struct)
+
+    case errors do
+      [] -> {:ok, struct}
+      _ -> {:error, errors}
+    end
   rescue
     e -> {:error, e}
   end
 
+  defp return_errors(struct) do
+    []
+    |> add_to_list_if_true({:error, :invalid_format} == DateTime.from_iso8601(struct.business.modifiedDate), [
+      %{"business.modifiedDate" => "Not ISO8601 formatted"}
+    ])
+  end
```

Comment:
I think leveraging pattern matching with `case` here makes it more readable. I'm not sure why we're returning a list of maps as the second element in the error tuple, but I've left it in case there's some context I'm missing.

```elixir
defp return_errors(struct) do
  case DateTime.from_iso8601(struct.business.modifiedDate) do
    {:error, :invalid_format} -> [%{"business.modifiedDate" => "Not ISO8601 formatted"}]
    _ -> []
  end
end
```

Reply:
I did it this way thinking that we'd be adding more requirements in the future and this could return a list of errors instead of just the first one.  Thoughts?

Reply:
That's an interesting idea. Leave the list then. Let's see if we like it.